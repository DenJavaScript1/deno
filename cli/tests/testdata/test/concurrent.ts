let value = 0;
async function counter(max: number) {
  value += 1;

  await new Promise((resolve) => {
    setTimeout(function recheck() {
      setTimeout(value == max ? resolve : recheck, 0);
    }, 0);
  });
}

const count = 10;
for (let i = 0; i < count; i++) {
  Deno.test({
    name: `counter ${i + 1}`,
    async fn() {
      return await counter(count);
    },
    concurrent: true,
  });
}
