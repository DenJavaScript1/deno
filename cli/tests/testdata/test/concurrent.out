Check [WILDCARD]/test/concurrent.ts
running 10 tests from [WILDCARD]/test/concurrent.ts
test counter 1 ... ok ([WILDCARD])
test counter 2 ... ok ([WILDCARD])
test counter 3 ... ok ([WILDCARD])
test counter 4 ... ok ([WILDCARD])
test counter 5 ... ok ([WILDCARD])
test counter 6 ... ok ([WILDCARD])
test counter 7 ... ok ([WILDCARD])
test counter 8 ... ok ([WILDCARD])
test counter 9 ... ok ([WILDCARD])
test counter 10 ... ok ([WILDCARD])

test result: ok. 10 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out ([WILDCARD])

